scripts/feature_tests/upgrade/combined_tests/1cp_1w_bootDiskImageANDK8sControllers_clusterLevel_upgrade.sh

```diff
@@ -0,0 +1,172 @@
+  log_test_result "upgrade_CAPI_and_CAPM3_with_clusterctl.sh" "fail"
+  exit 1
+fi
+
+sleep 30 # Wait for the controllers to be up and running
+
+health_controllers=$(kubectl get pods -A | grep -E "capm3-system|capi-kubeadm|metal3" | grep -vc 'Running')
+if [ "${health_controllers}" -ne 0 ]; then
+  log_error "Some of the upgraded controlplane componenets are not healthy"
+  log_test_result "upgrade_CAPI_and_CAPM3_with_clusterctl.sh" "fail"
+  exit 1
+fi
+
+# ----------- upgrade ironic image ------
+
+# Upgrade container images
+for container in ironic ironic-dnsmasq ironic-httpd mariadb; do
+  kubectl set image deployments metal3-ironic \
+    $container=quay.io/metal3-io/ironic:"${IRONIC_IMAGE_TAG}" -n metal3
+done
+kubectl set image deployments metal3-ironic \
+  ironic-inspector=quay.io/metal3-io/ironic-inspector:"${IRONIC_IMAGE_TAG}" -n metal3
+
+updated_ironic_image_count=$(kubectl get deployments metal3-ironic -n metal3 -o json |
+  jq '.spec.template.spec.containers[].image' | grep -c "${IRONIC_IMAGE_TAG}")
+
+if [ "${updated_ironic_image_count}" -lt 5 ]; then
+  echo "All ironic images are not updated properly"
+  exit 1
+fi
+
+sleep 60 # wait until images are downloaded
+
+# Check that ironic containers are running
+ironic_pod=$(kubectl get pods -n metal3 -o name | grep metal3-ironic)
+for container in ironic ironic-inspector ironic-dnsmasq ironic-httpd mariadb; do
+  running_containers_count=$(
+    kubectl get "${ironic_pod}" -n metal3 -o json |
+      jq ".status.containerStatuses[] | select(.name == \"${container}\") | .state" |
+      grep -ic running
+  )
+  echo "${running_containers_count}"
+  if [ "${running_containers_count}" -eq 0 ]; then
+    echo "Upgrade of ironic image for container ${container} has failed"
+    exit 1
+  fi
+done
+
+# ----------- upgrade boot disk image and kubernetes version ------
+echo "Create a new metal3MachineTemplate with new node image for both \
+controlplane and worker nodes"
+cp_Metal3MachineTemplate_OUTPUT_FILE="/tmp/cp_new_image.yaml"
+wr_Metal3MachineTemplate_OUTPUT_FILE="/tmp/wr_new_image.yaml"
+CLUSTER_UID=$(kubectl get clusters -n metal3 test1 -o json | jq '.metadata.uid' |
+  cut -f2 -d\")
+generate_metal3MachineTemplate "test1-new-controlplane-image" "${CLUSTER_UID}" \
+  "${cp_Metal3MachineTemplate_OUTPUT_FILE}"
+generate_metal3MachineTemplate "test1-new-workers-image" "${CLUSTER_UID}" \
+  "${wr_Metal3MachineTemplate_OUTPUT_FILE}"
+
+kubectl apply -f "${cp_Metal3MachineTemplate_OUTPUT_FILE}"
+kubectl apply -f "${wr_Metal3MachineTemplate_OUTPUT_FILE}"
+
+# controlloers
+kubectl get kcp -n metal3 test1 -o json |
+  jq ".spec.infrastructureTemplate.name=\"test1-new-controlplane-image\" | .spec.version=\"${UPGRADED_K8S_VERSION_2}\"" |
+  kubectl apply -f-
+kubectl get machinedeployment -n metal3 test1 -o json |
+  jq ".spec.strategy.rollingUpdate.maxSurge=1|.spec.strategy.rollingUpdate.maxUnavailable=0 | .spec.template.spec.version=\"${UPGRADED_K8S_VERSION_2}\"" |
```

Comment:
you could split this line, it is a bit long

Reply:
done